routes/users.js

```diff
@@ -0,0 +1,20 @@
+const { Router } = require('express');
+const router = Router();
+const pool = require('../db');
+
+router.get('/', (request, response) => {
+  pool.query('SELECT * FROM users ORDER BY id ASC', (err, res) => {
+    if (err) return next(err);
+    response.json(res.rows);
+  });
+});
+
+router.get('/:id', (request, response, next) => {
+  const { id } = request.params;
+  pool.query('SELECT * FROM users WHERE id = $1', [id], (err, res) => {
```

Comment:
I have a bad feeling about this. The client interacts directly with database, not a good way to do things.

Reply:
I don't follow. This is just like how a mongo query would be performed:

* Request comes to server from client
* Server queries database
* Server returns data to client

How else would this be handled?

Reply:
create APIs that the clients interacts with instead of querying the db

Reply:
Is this not the API? The Express routes? The `pool.query` is how the `pg` node package queries the postgres database. I assume it works similarly to `.findById` in Mongo.

I should just remove these queries for now anyways. They're just example filler, but I am still confused on your comment and be not confused would be ideal. 😄

Reply:
http://mherman.org/blog/2015/02/12/postgresql-and-nodejs/#server-side-routes you create an API route, client interacts with the predefined API route

Reply:
Here's the type of query I'm using in the examples in this file: https://node-postgres.com/features/pooling#single-query. It even says:

> This is the preferred way to query with node-postgres if you can as it removes the risk of leaking a client.

But I'm still learning how to use node-postgres and integrating it with some Express routes, so it's entirely possible I'm misunderstanding it all.